sdk/core/core/src/az_http_policy.c

```diff
@@ -5,12 +5,16 @@
 #include <az_http_pipeline.h>
 #include <az_http_policy.h>
 #include <az_http_request_builder.h>
+#include <az_log.h>
 #include <az_mut_span.h>
 #include <az_span.h>
 #include <az_span_builder.h>
 #include <az_str.h>
 #include <az_url.h>
 
+#include <stddef.h>
+#include <time.h>
+
 #include <_az_cfg.h>
 
 AZ_NODISCARD AZ_INLINE az_result az_http_pipeline_nextpolicy(
```

Comment:
This code isn't right. The logic should be like this:
- if should log (Request), create log entry with HTTP method, URL/path, query parameters and headers and then log it.
- Get current time and save it
- Call next policy
- Get current time and subtract previous time to get operation duration
- if should log (slow response) && duration > slowResponseThreshold, then log with HTTP response status line & response headers. Also, include the original HTTP method, url/path, QPs, and request headers
- if should log (Response), then log the same entry again (don't construct it twice though to save CPU).
- return from this policy

Reply:
Done, please check out the new code in this file, and in the az_log.c